src/config/utils.ts

```diff
@@ -27,3 +32,16 @@ export const checkIfDevtronOperatorHelmRelease = (
         clusterId === DEVTRON_DEFAULT_CLUSTER_ID
     )
 }
+
+/**
+ * Checks if the provided pathname matches the current path.
+ * If the paths do not match, returns a custom message or a default unsaved changes prompt.
+ *
+ * @param currentPathName - The current path to compare against.
+ * @param customMessage - Optional custom message to display when the path does not match.
+ * @returns A function that takes an object with a `pathname` property and performs the path match check.
+ */
+export const checkIfPathIsMatching =
```

Comment:
Can we set the return type from PromptProps, and if we are keeping it in dashboard ideally it should be in Helpers.tsx